### ISSUE TITLE
Items to the left of the Always Hidden bar do not hide 

### ISSUE THREAD

**Issue by kidpix** [dwarvesf/hidden#171]
### Describe the bug

Menu bar item to the left of the Always Hidden bar do not hide (see video).

### Steps to reproduce

1. Enable Always Hidden section in Preferences
2. Move menu bar item to the left of the translucent bar
3. Right click on the collapse button to hide

### Expected behaviour

Items to the left of the Always Hidden bar would be hidden

**Screenshots**

https://i.imgur.com/uHDNFCd.mp4

### Environment

OS: MacOS Big Sur 11.6

**Comment by vtmocanu** [comment]
Same behaviour here, also on Big Sur, M1 CPU.

**Comment by samanthafav** [comment]
Same here.
![Screen Shot 2021-12-06 at 2 18 24 PM](https://user-images.githubusercontent.com/20263743/144931815-648b7ac9-7da4-4f12-9b14-1a34b6e07644.png)

**Comment by gamedeviced** [comment]
Same here. Any plans on fixing this issue?

macOS Monterey 12.0.1
MacBook Pro 13-inch, 2018, Four Thunderbolt 3 Ports

Hidden Bar Version 1.8 (12)

**Comment by aayushjain** [comment]
So I've figured this one out. 
It's not a bug. Just an undocumented feature.

Screen-record by @kidpix is all correct, however the user at that time is just in a "configuration" mode. 
Hence the reason you are able to see the hide-line, and the always-hide-line.

To 'activate' the always hide part, there are two ways:
1. Right click on the expand-arrow, OR
2. Single-click on the exapand-arrow while holding down Option+Command.

Rearrange your icons in the hide and always hide sections, then make the hiding active.

Cheers! 🍻

**Comment by gamedeviced** [comment]
@aayushjain Nope, @kidpix rightclicks the arrow to hide the lines as you see in the last part of the video. The lines disappear but the icon left to the hide line is still visible. Or am I getting something wrong?

**Comment by viettrungphan** [comment]
Hi, can you try the new release, I clean up and rewrite the logic for 2 separated buttons. The old code is messing up may cause many side effects.
You can check the new build here, the AppStore need more time to submit.
https://github.com/dwarvesf/hidden/releases/tag/v1.9

**Comment by gamedeviced** [comment]
@viettrungphan Somethings seems wrong with this version, at least for me. Right clicking the arrow does nothing and icons left to the hide line are still visible.

**Comment by viettrungphan** [comment]
Please make sure the "Alway hide feature" enabled in preferences window. Right click only use for that feature. Left click using to show hide normal areas.

**Comment by gamedeviced** [comment]
@viettrungphan It is activated. Here's a screen recording of v1.9. In the last part you see me trying to right click on my trackpad. Thats why I sometimes do a left click. But especially when you see the arrow background highlighted that's a right click. You can see that he left line (hide line) disappeary but the right line stays visible.

Edit: lol! After this I downgraded to 1.8 and the features works. I dunno what's going on sorry …

https://user-images.githubusercontent.com/70090026/150541965-5ce2284b-489b-4562-9794-a7c15cde0c48.mov

**Comment by viettrungphan** [comment]
Can you delete and reinstall the app again. Currently I unable to reproduce the bug yet.

**Comment by mlcampbe** [comment]
I think I may have figured this out at least with the 1.9 verison. You have to open the preferences and check the box for "enable always hidden section" so that the transparent line appears. Then close the preferences and exit the app completely. Restart the app and then move the desired icons to the left of the transparent bar. At this point you can right click the arrow icon to toggle the always hidden icons on/off. At least that works for me.

**Comment by wsmbsbbz** [comment]
> I think I may have figured this out at least with the 1.9 verison. You have to open the preferences and check the box for "enable always hidden section" so that the transparent line appears. Then close the preferences and exit the app completely. Restart the app and then move the desired icons to the left of the transparent bar. At this point you can right click the arrow icon to toggle the always hidden icons on/off. At least that works for me.

It works for me! @viettrungphan Maybe you can find some clues from it?

**Comment by viettrungphan** [comment]
@wsmbsbbz I rewrite the logic of menu bar buttons in version 1.9, the old code mess up and easy to cause error happened, I suggest everybody try it. We don't have QC for it that is the reason why I release it as a "Prerelease" tag.

**Comment by peerasut** [comment]
> @viettrungphan It is activated. Here's a screen recording of v1.9. In the last part you see me trying to right click on my trackpad. Thats why I sometimes do a left click. But especially when you see the arrow background highlighted that's a right click. You can see that he left line (hide line) disappeary but the right line stays visible.
> 
> Edit: lol! After this I downgraded to 1.8 and the features works. I dunno what's going on sorry …
> 
>  Bildschirmaufnahme.2022-01-21.um.15.12.38.mov

Same to me just downgraded to 1.8, it just works! (on macOS 12.3.1)

**Comment by LPFchan** [comment]
Hidden Bar 1.8 (App Store) on macOS 12.4 this is still happening. 

Edit: Updated to 1.9 from GitHub and it's now fixed!

**Comment by belumira** [comment]
If I open preferences of a hidden icon, it closes when Hidden Bar automatically hides icons

**Comment by d4vidsha** [comment]
Also happens when first installing without a computer restart. A restart fixed my issue.

**Comment by thachtranvn** [comment]
Same issue. I already tried to re-install v1.9, but it doesn't work. 
@viettrungphan I'm living in HCMC, so I can help you to reproduce the bug :))

**Comment by mrdc** [comment]
> Hidden Bar 1.8 (App Store) on macOS 12.4 this is still happening.
> 
> Edit: Updated to 1.9 from GitHub and it's now fixed!

For me 1.8 from App Store works fine on Ventura 13.6. Just follow the instruction above: close and then re-open the app after enabling always hide. 
![image](https://github.com/dwarvesf/hidden/assets/16876990/c888f67f-7ce4-4d6c-84a3-9c3f5ddd338c)

**Comment by mrdc** [comment]
> Same issue. I already tried to re-install v1.9, but it doesn't work. @viettrungphan I'm living in HCMC, so I can help you to reproduce the bug :))

1.8 from App Store is working fine - just use the instruction in this issue.

**Comment by shu-ming** [comment]
after setting always hidden, only works when relaunching the app.

**Comment by elidub** [comment]
> after setting always hidden, only works when relaunching the app.

How do I relaunch the app? I couldn't find any way to do this (besides restarting the computer).

**Comment by wsmbsbbz** [comment]
> > after setting always hidden, only works when relaunching the app.
> 
> How do I relaunch the app? I couldn't find any way to do this (besides restarting the computer).

You can try with Activity Monitor.

**Comment by elidub** [comment]
> > > after setting always hidden, only works when relaunching the app.
> > 
> > How do I relaunch the app? I couldn't find any way to do this (besides restarting the computer).
> 
> You can try with Activity Monitor.

Yes, that works! Thank you!

**Comment by jared-christensen** [comment]
Shoutout to @mlcampbe for originally finding the workaround in 2022—it’s still working in 2024!

You'll need to restart Hidden Bar after enabling **Always Hidden Section** for the changes to take effect. Here's what to do:

1. Open **Hidden Bar** and enable **Always Hidden Section** (a transparent bar will appear).
2. Quit Hidden Bar using **Activity Monitor** (find it, hit the **X**).
3. Reopen **Hidden Bar** and move icons left of the transparent bar.
4. Right-click the **arrow icon** to hide the bars and the hidden apps.
